docs/blog/posts/03-deploy-llms-on-a-budget.md

```diff
@@ -0,0 +1,319 @@
+| [phi-2](https://huggingface.co/microsoft/phi-2) | 27.98 | 70.8 | 44.43 | 35.21 | 44.61 |
+
+
+## 💸 Serving Phixtral on a budget with SkyPilot and NOS
+
+Let's now see how we can deploy the [mlabonne/phixtral-4x2_8](https://huggingface.co/mlabonne/phixtral-4x2_8) model on a single NVIDIA L4 GPU for under $160 / month. We'll be using [SkyPilot](https://skypilot.readthedocs.io/) to deploy and manage our [NOS](https://github.com/autonomi-ai/nos) service on spot (pre-emptible) instances, making them especially cost-efficient.
+
+!!!question "What's SkyPilot?"
+    If you're new to SkyPilot, we recommend you go through our [SkyPilot integration page](/docs/integrations/skypilot.md) first to familiarize yourself with the tool.
+
+### 1. Define your custom model and serve specification
+
+In this example, we'll be using the [`llm-streaming-chat`](https://github.com/autonomi-ai/nos-playground/blob/main/examples/llm-streaming-chat/) tutorial on [NOS playground](https://github.com/autonomi-ai/nos-playground). First, we'll define our custom phixtral chat model [`phixtral_chat.py`](https://github.com/autonomi-ai/nos-playground/blob/main/examples/llm-streaming-chat/models/phixtral_chat.py) and a [`serve.phixtral.yaml`](https://github.com/autonomi-ai/nos-playground/blob/main/examples/llm-streaming-chat/serve.phixtral.yaml) serve specification that will be used by NOS to serve our model. The relevant files are shown below:
+
+```bash
+(nos-py38) nos-playground/examples/llm-streaming-chat $ tree .
+├── models
+│   └── phixtral_chat.py
+├── serve.phixtral.yaml
+```
+
+The entire chat interface is defined in the `StreamingChat` module in [`phixtral_chat.py`](https://github.com/autonomi-ai/nos-playground/blob/main/examples/llm-streaming-chat/models/phixtral_chat.py), where the `chat` method returns a string iterable for the gRPC / HTTP server to stream back model predictions to the client. 
+
+The [`serve.phixtral.yaml`](https://github.com/autonomi-ai/nos-playground/blob/main/examples/llm-streaming-chat/serve.phixtral.yaml) serve specification defines the custom chat model, and a custom runtime that NOS uses to execute our model. Follow the annotations below to understand the different components of the serve specification.
+
+<div class="annotate" markdown>
+
+```yaml title="serve.phixtral.yaml"
+
+images: (1)
+  llm-py310-cu121: (2)
+    base: autonomi/nos:latest-py310-cu121 (3)
+    pip: (4)
+      - bitsandbytes
+      - transformers
+      - einops
+      - accelerate
+
+models: (5)
+  mlabonne/phixtral-4x2_8: (6)
+    model_cls: StreamingChat (7)
+    model_path: models/phixtral_chat.py (8)
+    init_kwargs:
+      model_name: mlabonne/phixtral-4x2_8
+    default_method: chat
+    runtime_env: llm-gpu
+    deployment: (9)
+      resources:
+        device: auto
+        device_memory: 7Gi (10)
+```
+</div>
+1. Specifies the custom runtime images that will be used to serve our model.
```

Comment:
Maybe mention that this is a NOS image? still want to emphasize that this is all running in NOS